### ISSUE TITLE
Edge this.userId set to null

### ISSUE THREAD

**Issue by Kaczkazniszczenia** [VeliovGroup/Meteor-Files#360]
While using Edge 38.14393.0.0 we are facing an issue where context of options "protect" and "downloadCallback" has this.userId === null and this.user() === null.

## Demo code
We have created a small demo app showing an issue: https://bitbucket.org/CallMeMB/files-test 

Demo will return false from downloadCallback if userId is not set

## How to reproduce:
1. Start demo app
2. Load application in Edge
3. Login into user account
4. Download file (it will be successfull)
5. Reload the page (ctrl+r)
6. Download file (it will fail)

## Expected result:
File will always be downloaded

## Versions:
Meteor: 1.4.2.7

### Packages: 
- ostrio:files@1.7.11
- ostrio:cookies@2.1.3
- accounts-base@1.2.14
- accounts-password@1.3.4

### Npm packages:
- gridfs-stream 1.1.1

### Server OS:
Windows 10 Professional (development machine, localhost)
Linux (docker container, test/production env)

**Comment by dr-dimitru** [comment]
Hello @Kaczkazniszczenia ,

 1. What happens between steps 5 and 6, why user is not logged-in?
 2. Could you obtain request headers (*including cookies*) on steps 4 and 6, and compare (*diff*) it? What the difference?

Meanwhile you should take a look at [this code](https://github.com/thomasspiesser/ostrio-files-bug/pull/1/files), may be authorization via cookies isn't suitable for you and you would like to use gwt query.

**Comment by Kaczkazniszczenia** [comment]
Hello @dr-dimitru ,

There is nothing between 4 and 5. Reload the page and click the download link.
Meteor see user as logged in all time.

Headers looks exactly the same (on success and failure). 
```
   Accept: */*
   Accept-Encoding: gzip, deflate
   Connection: Keep-Alive
   Cookie: x_mtok=FoYb3fehJZyGFGEfo
   Host: localhost:3000
   Referer: http://localhost:3000/
   User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.79 Safari/537.36 Edge/14.14393
```

Hack with get parameter (shown in link) is working but I don't see it as a proper solution, more like temporary workaround since this.userId work correctly on other browsers.

BTW. no issues on other browsers.

**Comment by dr-dimitru** [comment]
Value of `x_mtok` is new on every login. Is there a chance this cookie remains the same between page refresh? Also life-time of this cookie is session since `v1.7.7`

What meteor and package version you're on?

**Comment by Kaczkazniszczenia** [comment]
I've set up testing server under http://files.test.oknonews.com/ 
### How to use:
- You can login using account displayed in top left corner
- context of last (on server) file request is displayed under login ui
- context is refreshing in protect() option
- each file is downloadable using two interfaces 'Param send ...' and 'Redirect ...' (more below)
- to use redirect ui you need to be logged in

### Request context:
- User context id: userId got from cookie (default Meteor-files behavior)
- User x-mtok id: userId got using Meteor.server.sessions[ **Client param x-mtok** ] 
- Client param x-mtok: session token got from "Meteor.connection._lastSessionId" and send by url param
- Server sessions: list of active sessions

### Download interface:
- Param send ... : solution used from https://github.com/thomasspiesser/ostrio-files-bug/pull/1/files
- Redirect ... : similar to "Param send" but appends x-mtok differently. File download url is showing on link were user is redirected to file.link() with x-mtok glued. (again, blocked for not logged in users)

### Observations:
- after page refresh **User context id** is lost
- if using Redirect interface, there is strong possibility  **Server sessions** wont contain **Client param x-mtok**

**Comment by Kaczkazniszczenia** [comment]
Meteor-files show same tendensy as Redirect interface described before.
file **files.coffe**, line 1177, function _getUser
After refreshing page, value mtok from cookie tends to not match any of Meteor.server.sessions values, so in line 1194 if statment returns false.

**Comment by Kaczkazniszczenia** [comment]
I have found root of a problem.
Accounts.onLogin event does not trigger on page reload (only on Misscossoft Edge)
I fixed it using Meteor.startup.

I would like you to review this pull request, if there is something wrong please notify me about it.

**Comment by dr-dimitru** [comment]
PR is merged and package is published as [`v1.7.12`](https://github.com/VeliovGroup/Meteor-Files/releases/tag/1.7.12), thank you for contribution
